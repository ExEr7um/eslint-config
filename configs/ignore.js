import gitignore from "eslint-config-flat-gitignore"

/** @type {import('eslint').Linter.FlatConfig} */
export default [
  gitignore({
    root: true,
  }),

  {
    ignores: ["package-lock.json", "pnpm-lock.yaml", "yarn.lock"],
    name: "general/ignore",
  },
]
