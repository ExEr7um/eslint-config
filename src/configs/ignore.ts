import type { Linter } from "eslint"

import gitignore from "eslint-config-flat-gitignore"

export default [
  gitignore({
    root: true,
  }),

  {
    ignores: ["**/package-lock.json", "**/pnpm-lock.yaml", "**/yarn.lock"],
    name: "general/ignore",
  },
] as const satisfies Linter.Config[]
