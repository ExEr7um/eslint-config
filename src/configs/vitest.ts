import type { Linter } from "eslint"

import vitest from "@vitest/eslint-plugin"

export default [
  {
    files: ["**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts}"],
    name: "vitest/base",
    plugins: { vitest },
    rules: {
      ...vitest.configs.recommended.rules,
      "vitest/consistent-test-filename": [
        "error",
        { pattern: String.raw`.*\.spec\.ts` },
      ],
      "vitest/consistent-test-it": ["error", { fn: "test" }],
      "vitest/no-alias-methods": "error",
      "vitest/no-conditional-expect": "error",
      "vitest/no-conditional-in-test": "error",
      "vitest/no-conditional-tests": "error",
      "vitest/no-disabled-tests": "error",
      "vitest/no-focused-tests": ["error", { fixable: false }],
      "vitest/no-interpolation-in-snapshots": "error",
      "vitest/prefer-comparison-matcher": "error",
      "vitest/prefer-each": "error",
      "vitest/prefer-equality-matcher": "error",
      "vitest/prefer-hooks-in-order": "error",
      "vitest/prefer-hooks-on-top": "error",
      "vitest/prefer-strict-equal": "error",
      "vitest/prefer-to-be": "error",
      "vitest/prefer-to-be-falsy": "error",
      "vitest/prefer-to-be-object": "error",
      "vitest/prefer-to-be-truthy": "error",
      "vitest/prefer-to-contain": "error",
      "vitest/prefer-to-have-length": "error",
      "vitest/require-top-level-describe": "error",
    },
  },
] as const satisfies Linter.Config[]
