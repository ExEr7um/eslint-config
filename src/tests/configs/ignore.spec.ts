import { describe, expect, test } from "vitest"

import ignore from "../../configs/ignore"
import createESLintConfig from "../../index"

describe("Конфиг ignore", () => {
  test("Включен по умолчанию", async () => {
    const config = await createESLintConfig()

    expect(config).toStrictEqual(expect.arrayContaining(ignore))
  })

  test("Имеет все параметры", () => {
    expect(ignore).toMatchSnapshot()
  })
})
