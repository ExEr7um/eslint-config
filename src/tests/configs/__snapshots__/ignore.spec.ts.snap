// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Конфиг ignore > Имеет все параметры 1`] = `
[
  {
    "ignores": [
      "**/node_modules",
      "**/*.log*",
      "**/.nuxt",
      "**/.nitro",
      "**/.cache",
      "**/.output",
      "**/.env",
      "**/dist",
      "**/.idea/",
      "**/.eslintcache",
      "**/coverage",
    ],
    "name": "gitignore",
  },
  {
    "ignores": [
      "**/package-lock.json",
      "**/pnpm-lock.yaml",
      "**/yarn.lock",
      "**/bun.lock",
    ],
    "name": "general/ignore",
  },
]
`;
